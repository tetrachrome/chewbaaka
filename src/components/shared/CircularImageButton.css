/**
 * CircularImageButton.css
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 30, 2020
 */

div.CircularImageButtonOuterContainer {
  display: inline-block;
}

div.CircularImageButtonOuterContainer_DebugColor {
  background-color: gray;
}

div.CircularImageButtonCore {
  width:212px;
  height:212px;
  border:6px solid orange;  
  border-radius:50%;
  -moz-border-radius:50%;
  -webkit-border-radius:50%;  
}

img.CircularImageButtonImg {
  width:200px;
  height:200px;
  border-radius: 50%;
}

img.CircularImageButtonImg_DebugColor {
  background-color: pink;
}

img.CircularImageButtonImg:hover {
  filter: contrast(120%);
}

p.CircularImageButtonTitle {
  width: 212px;
  margin-top: 5px;
  text-align: center;
  font-size: 22px;
  /* font-weight: 800; */
}

p.CircularImageButtonTitle_DebugColor {
  background-color: gold;
}
