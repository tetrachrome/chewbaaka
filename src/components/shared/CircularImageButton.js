/**
 * CircularImageButton.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 06, 2020
 */

import React from 'react';

import "semantic-ui-css/semantic.min.css";

import './CircularImageButton.css'

export default class CircularImageButton extends React.Component {

  render() {
    return (
      <div className="CircularImageButtonOuterContainer">
        <a href={this.props.href}>
          <div className="CircularImageButtonCore">
            <img className="CircularImageButtonImg" src={this.props.image} alt={this.props.title} />
          </div>
          <div>
            <p className="CircularImageButtonTitle">
              {this.props.title}
            </p>
          </div>
        </a>
      </div>
    );
  }
}
