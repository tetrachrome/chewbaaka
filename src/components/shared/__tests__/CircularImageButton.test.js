/**
 * CircularImageButton.test.js
 * Chewbaaka
 *
 * Author   : Tomiko
 * Created  : Jul 06, 2020
 * Updated  : Jul 08, 2020
 */

import React from 'react';

import { render, screen } from '@testing-library/react';

import renderer from 'react-test-renderer';

import CircularImageButton from '../CircularImageButton';

const title = "Cheetahology";
const image = "https://cheetah.org/wp-content/uploads/2019/01/cheetah-conservation-fund-cheetah-fact-01.jpg";
const href  = "https://cheetah.org/wp-content/uploads/2019/01/cheetah-conservation-fund-cheetah-fact-01.jpg";

test('renders CircularImageButton component', () => {
  render(
    <CircularImageButton
      title={title}
      image={image}
      href={href}
    />
  );

  const titleElement = screen.getByText(title);
  expect(titleElement).toBeInTheDocument();
});

test('CircularImageButton component snapshot', () => {
  const tree = renderer
    .create(
      <CircularImageButton
        title={title}
        image={image}
        href={href}
      />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
