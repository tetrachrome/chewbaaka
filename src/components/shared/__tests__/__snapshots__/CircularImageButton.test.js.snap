// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CircularImageButton component snapshot 1`] = `
<div
  className="CircularImageButtonOuterContainer CircularImageButtonOuterContainer_DebugColor DebugWireframe"
>
  <a
    href="https://cheetah.org/wp-content/uploads/2019/01/cheetah-conservation-fund-cheetah-fact-01.jpg"
  >
    <div
      className="CircularImageButtonCore"
    >
      <img
        alt="Cheetahology"
        className="CircularImageButtonImg CircularImageButtonImg_DebugColor DebugWireframe"
        src="https://cheetah.org/wp-content/uploads/2019/01/cheetah-conservation-fund-cheetah-fact-01.jpg"
      />
    </div>
    <div>
      <p
        className="CircularImageButtonTitle CircularImageButtonTitle_DebugColor DebugWireframe"
      >
        Cheetahology
      </p>
    </div>
  </a>
</div>
`;
